### DISCUSSION TITLE
PHP: Implement OpenTelemetry in runtime

### DISCUSSION THREAD

**Discussion by charjr** [dagger/dagger#12270] [maintainers]
Make use of OpenTelemetry in PHP SDK runtime so we can get real-time feedback as opposed receiving all output only once complete.

- [X] Show step details in TUI
- [ ] Implement Otel within SDK
- [ ] Use custom spans
  - More detail in between steps to improve debugging experience and so on.

**Comment by charjr** [comment]
@wingyplus You've added telemetry to the PHP SDK. Is there anything missing or is this issue solved?

**Comment by wingyplus** [comment]
What I do is making the SDK to show correct information on the TUI by setting the traceparent to the http header. You can see a conversation in https://github.com/dagger/dagger/pull/8859.

one thing is missing here is custom span. If you see in the Go SDK, you can call Tracer function and the start a span with custom message. This part needs the SDK to have OTel to implements a  custom exporter.
